src/libraries/System.Net.Http/tests/FunctionalTests/DiagnosticsTests.cs

```diff
@@ -46,43 +46,36 @@ public void EventSource_ExistsWithCorrectId()
         // could interfere with the tests, as well as the enabling of tracing interfering
         // with those tests.
 
-        /// <remarks>
-        /// This test must be in the same test collection as any others testing HttpClient/WinHttpHandler
-        /// DiagnosticSources, since the global logging mechanism makes them conflict inherently.
-        /// </remarks>
-        [OuterLoop("Uses external servers")]
         [ConditionalFact(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
         public void SendAsync_ExpectedDiagnosticSourceLogging()
         {
-            RemoteExecutor.Invoke(useVersionString =>
+            RemoteExecutor.Invoke(async (useVersion, testAsync) =>
             {
-                bool requestLogged = false;
+                HttpRequestMessage requestLogged = null;
+                HttpResponseMessage responseLogged = null;
                 Guid requestGuid = Guid.Empty;
-                bool responseLogged = false;
                 Guid responseGuid = Guid.Empty;
                 bool exceptionLogged = false;
                 bool activityLogged = false;
 
+                SemaphoreSlim responseLoggedSemaphore = new(0, 1);
```

Comment:
Since you are just using this as an async signal, it might be simpler to use a TaskCompletionSource.